components/ProgressView/src/MDCProgressView.m

```diff
@@ -292,7 +292,9 @@ + (UIColor *)defaultTrackTintColorForProgressTintColor:(UIColor *)progressTintCo
 
 - (void)updateProgressView {
   // Update progressView with the current progress value.
-  CGFloat progressWidth = MDCCeil(self.progress * CGRectGetWidth(self.bounds));
+  CGFloat scale = self.window.screen ? self.window.screen.scale : 1;
+  CGFloat width = self.progress * CGRectGetWidth(self.bounds);
+  CGFloat progressWidth = MDCRound(width * scale) / scale;
```

Comment:
Did you accidentally switch from ceiling to rounding behavior?  MDCProgressView has been using ceiling functions since 2016 and I think switching should be part of a different PR.  I think it's best to keep this one focused on the pixel-alignment issue.

I think `MDCCeilScaled` would be helpful here, but you could use `MDCCeil` if you don't like the scaled implementation.

Reply:
I did change it intentionally in an effort to fix some surprising behavior, but I agree that we can approach that separately so for now I've added some unit tests to demonstrate the surprising behavior I encountered.

Reply:
*and reverted this back to ceil for now.

Reply:
Note the new `test97` tests in the new unit tests and the fact that the frame width is actually 98 on 1pt displays.

Reply:
Filed https://github.com/material-components/material-components-ios/issues/9482 to track.